hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncRegionLocatorHelper.java

```diff
@@ -67,7 +68,8 @@ static void updateCachedLocationOnError(HRegionLocation loc, Throwable exception
       LOG.debug("The actual exception when updating {} is {}", loc,
         cause != null ? cause.toString() : "none");
     }
-    if (cause == null || !isMetaClearingException(cause)) {
+    if (cause == null || !isMetaClearingException(cause)
```

Comment:
So now we need to modify the code other than ClientExceptionsUtil, this means we need to create patch for branch-2? As on branch-2, I believe there are other code paths which need to handle this...

Reply:
Yeah, I create another pull request to branch2: https://github.com/apache/hbase/pull/578

Reply:
@johnhomsea  You created a new PR. Does it subsume this one? Or is it complementary? Thanks.

Reply:
Yes, it subsumes this one, but branch-2 is different from master in some cases.